test-programs/compressed-token-test/tests/test.rs

```diff
@@ -237,27 +242,418 @@ async fn test_mint_to<const MINTS: usize, const ITER: usize>() {
+    // Make sure that the total token supply does not exceed `u64::MAX`.
+    let amounts: Vec<u64> = (0..20).map(|_| rng.gen_range(0..(u64::MAX / 20))).collect();
+    test_mint_to(amounts, 1).await
 }
 
 #[tokio::test]
 async fn test_25_mint_to() {
-    test_mint_to::<25, 10>().await
+    let mut rng = rand::thread_rng();
+    // Make sure that the total token supply does not exceed `u64::MAX`.
+    let amounts: Vec<u64> = (0..25)
+        .map(|_| rng.gen_range(0..(u64::MAX / (25 * 10))))
+        .collect();
+    test_mint_to(amounts, 10).await
+}
+
+/// Failing tests:
+/// 1. Try to mint token from `mint_1` and sign the transaction with `mint_2`
+///    authority.
+/// 2. Try to mint token from `mint_2` and sign the transaction with `mint_1`
+///    authority.
+/// 3. Try to mint token from `mint_1` while using `mint_2` pool.
+/// 4. Try to mint token from `mint_2` while using `mint_1` pool.
+/// 5. Invalid CPI authority.
+/// 6. Invalid registered program.
+/// 7. Invalid noop program.
+/// 8. Invalid account compression authority.
+/// 9. Invalid Merkle tree.
+/// 10. Mint more than `u64::MAX` tokens.
+/// 11. Multiple mints which overflow the token supply over `u64::MAX`.
+#[tokio::test]
+async fn test_mint_to_failing() {
+    const MINTS: usize = 10;
+
+    let (mut rpc, env) = setup_test_programs_with_accounts(None).await;
+    let payer_1 = rpc.get_payer().insecure_clone();
+    let merkle_tree_pubkey = env.merkle_tree_pubkey;
+
+    let mut rng = rand::thread_rng();
+
+    let payer_2 = Keypair::new();
+    airdrop_lamports(&mut rpc, &payer_2.pubkey(), 1_000_000_000)
+        .await
+        .unwrap();
+
+    let mint_1 = create_mint_helper(&mut rpc, &payer_1).await;
+    let mint_pool_1 = get_token_pool_pda(&mint_1);
+
+    let mint_2 = create_mint_helper(&mut rpc, &payer_2).await;
+    let mint_pool_2 = get_token_pool_pda(&mint_2);
+
+    // Make sure that the tokal token supply does not exceed `u64::MAX`.
+    let amounts: Vec<u64> = (0..MINTS)
+        .map(|_| rng.gen_range(0..(u64::MAX / MINTS as u64)))
+        .collect();
+    let recipients = amounts
+        .iter()
+        .map(|_| Keypair::new().pubkey())
+        .collect::<Vec<_>>();
+
+    let instruction_data = light_compressed_token::instruction::MintTo {
+        amounts: amounts.clone(),
+        public_keys: recipients.clone(),
+    };
+
+    // 1. Try to mint token from `mint_1` and sign the transaction with `mint_2`
+    //    authority.
+    {
+        let instruction = create_mint_to_instruction(
+            &payer_2.pubkey(),
+            &payer_2.pubkey(),
+            &mint_1,
+            &merkle_tree_pubkey,
+            amounts.clone(),
+            recipients.clone(),
+        );
+        let result = rpc
+            .create_and_send_transaction(&[instruction], &payer_2.pubkey(), &[&payer_2])
+            .await;
+        assert_rpc_error(
+            result,
+            0,
+            anchor_lang::error::ErrorCode::ConstraintRaw.into(),
```

Comment:
We can add error codes in the constraints, I think we should do that to be sure that the correct constraint fails here.
Example from `register_program.rs`
``    #[account(mut, constraint= authority.key() == group_authority_pda.authority @AccountCompressionErrorCode::InvalidAuthority)]``

Reply:
Done